### ISSUE TITLE
macOS Monterey UI glitch on show Toolbar

### ISSUE THREAD

**Issue by sweetppro** [transmission/transmission#3328]
There is a slight UI glitch when showing the Toolbar on Monterey <edit>when aurosize is enabled</edit>, this is due to the Unified Toolbar used...
If we changed to the older style Toolbar like used in Mojave the render issue can be easily fixed.

However its maybe good to eat the UI issue, and hope Apple fixes it, for a better feel for Transmission on Monterey

### Current Toolbar on macOS Monterey:
<img width="650" alt="Screen Shot 2022-06-20 at 8 44 15 am" src="https://user-images.githubusercontent.com/7323792/174541020-489e09bf-38e6-489f-be12-0d97ccab361c.png">

### Expanded Toolbar on macOS Monterey:
<img width="650" alt="Screen Shot 2022-06-20 at 8 50 10 am" src="https://user-images.githubusercontent.com/7323792/174541891-165ffb74-eea5-405f-ac36-ee5756bf6bd2.png">

#3327 #2888

**Comment by sweetppro** [comment]
@Coeur ate you running big sur or monterey? The bug occurs when toggling the toolbar to visible. The window “jumps” to the correct size

**Comment by sweetppro** [comment]
The thing is, I do prefer the new look, and would prefer to find a work around.

**Comment by sweetppro** [comment]
Interesting,
I see the glitch on my M1 Air. Perhaps it’s only on Apple Silicon

**Comment by sweetppro** [comment]
@Coeur Ah wait, you need to enable autosize

**Comment by Coeur** [comment]
Same. Waiting for your video recording. :)

**Comment by sweetppro** [comment]
here it is:

https://user-images.githubusercontent.com/7323792/174878959-d95d2695-2321-4e9a-8a46-fd15ee4ad1a2.mov

**Comment by nevack** [comment]
Isn't it just expected? Like what is AppKit supposed to do?

**Comment by sweetppro** [comment]
the animation should be smooth, theres a slight jump when showing the toolbar.
its not as obvious in the video for some reason.
I guess the frame rate is too low

**Comment by nevack** [comment]
> the animation should be smooth, theres a slight jump when showing the toolbar. its not as obvious in the video for some reason. I guess the frame rate is too low

But is it smooth if you create some test app that just toggles toolbar visibility?

**Comment by sweetppro** [comment]
try this, create a new project.
add a toolbar to the window
add a view to the window
use the following constraints on the view:
* Top Space to superview = 0
* Bottom space to superview = 0
* Leading space to superview = 0
* Trailing space to superview = 0
* height = 200 (or any constant value)

run the app, and toggle the toolbar.
it will crash when you hide, then try to show the toolbar.

This is what this code tries to get around:
```
- (void)updateWindowAfterToolbarChange
{
    //Hacky way of fixing an issue with showing the Toolbar
    if (!self.isFullScreen && [self.fDefaults boolForKey:@"AutoSize"])
    {
        //macOS Big Sur shows the unified toolbar by default
        //and we only need to "fix" the layout when showing the toolbar
        if (@available(macOS 11.0, *))
        {
            if (!self.fWindow.toolbar.isVisible)
            {
                [self removeStackViewHeightConstraints];
            }
        }
        else
        {
            [self removeStackViewHeightConstraints];
        }

        dispatch_async(dispatch_get_main_queue(), ^{
            [self setWindowSizeToFit];
        });
    }
}
```
This code essentially removes the height constraint, the reapplies it.
Because of this, we get the UI glitch when restoring the height constraint.
*This is only an issue when autosize is enabled, since we want the window height to remain constant*

This is an issue with how Apple changes the content height when toggling the toolbar.

**Comment by ckerr** [comment]
> Can you try again with latest nightly?
